WooCommerce/Classes/CardPresentPayments/CardPaymentsAdaptor.swift

```diff
@@ -0,0 +1,147 @@
+        let preflightController = await CardPresentPaymentPreflightController(
+            siteID: siteID,
+            configuration: CardPresentConfigurationLoader().configuration,
+            rootViewController: UIViewController(),
+            alertsPresenter: self,
+            onboardingPresenter: self,
+            bluetoothConnectionController: bluetoothConnectionController,
+            builtInConnectionController: tapToPayConnectionController)
+        let orderPaymentUseCase = await CollectOrderPaymentUseCase(siteID: siteID,
+                                                                   order: order,
+                                                                   formattedAmount: currencyFormatter.formatAmount(order.total, with: order.currency) ?? "",
+                                                                   // moved from EditableOrderViewModel.collectPayment(for: Order)
+                                                                   rootViewController: UIViewController(),
+                                                                   // We don't want to use this at all, but it's currently required by the existing code.
+                                                                   // TODO: replace `rootViewController` with a protocol containing the UIVC functions we need, and implement that here.
+                                                                   onboardingPresenter: self,
+                                                                   configuration: CardPresentConfigurationLoader().configuration,
+                                                                   alertsPresenter: self,
+                                                                   preflightController: preflightController)
+        return await withCheckedContinuation { continuation in
+            orderPaymentUseCase.collectPayment(using: discoveryMethod) { error in
+                if let error = error as? CardPaymentErrorProtocol {
+                    continuation.resume(returning: .failure(error))
+                } else {
+                    continuation.resume(returning: .failure(CardPaymentsAdaptorError.unknownPaymentError(underlyingError: error)))
+                }
+            } onCancel: {
+                continuation.resume(returning: .cancellation)
+            } onPaymentCompletion: {
+                // no-op – not used in PaymentMethodsViewModel anyway so this can be removed
+            } onCompleted: {
+                continuation.resume(returning: .success(order))
+            }
+        }
+    }
+
+    enum CardPaymentsAdaptorError: Error, CardPaymentErrorProtocol {
+        var retryApproach: CardPaymentRetryApproach {
+            .restart
+        }
+
+        case unknownPaymentError(underlyingError: Error)
+    }
+}
+
+extension CardPresentPaymentsAdaptor: CardPresentPaymentsOnboardingPresenting, CardPresentPaymentAlertsPresenting {
```

Comment:
Will the adaptor also split responsibilities here along `*_OnboardingPresenting` and `*_AlertsPresenting`? Or we keep all in one?

Reply:
I don't have a strong feeling on splitting them. We should do it if we found that it made it easier to use at the call site, e.g. because the SwiftUI view modifiers we use to present each were different.

For the purposes of the experiment, it doesn't make a lot of difference though.